flow/layers/layer.h

```diff
@@ -34,11 +36,24 @@ namespace testing {
 class MockLayer;
 }  // namespace testing
 
+class PictureLayer;
+class DisplayListLayer;
+class PerformanceOverlayLayer;
+class TextureLayer;
+
 static constexpr SkRect kGiantRect = SkRect::MakeLTRB(-1E9F, -1E9F, 1E9F, 1E9F);
 
 // This should be an exact copy of the Clip enum in painting.dart.
 enum Clip { none, hardEdge, antiAlias, antiAliasWithSaveLayer };
 
+struct RasterCacheEntry {
+  explicit RasterCacheEntry(Layer* layer)
+      : layer(layer), num_child_entries(0), need_caching(false) {}
+
+  Layer* layer;
```

Comment:
I see that you handle this now by having each layer implement a method to prepare the cache. That works too. Perhaps consider having that be a pure virtual class for them to inherit from rather than baking it into Layer and having layers that don't care about caching having a method they will never use. You can also type this as "CacheableLayer*" which enforces that layers that want to cache will need to inherit from that pure virtual class and implement the method.